### PR TITLE
Disable HIP code where ROCm is disabled

### PR DESCRIPTION
**Pull request by Rombur**
This allows me to compile a few more serial tests. I can't compile all of them yet

Blocked by #15

### CODE REVIEW

**Comment by Rombur** [comment]
**ping** @dalg24

**Comment by Rombur** [comment]
If there is no comment on this by tomorrow, I will merge this. I am not going to ping you everyday on every PR I write.

File: core/src/impl/Kokkos_Memory_Fence.hpp [hunk 1]
```diff
@@ -54,6 +54,8 @@ void memory_fence() {
   __threadfence();
 #elif defined(KOKKOS_ENABLE_ROCM_ATOMICS)
   amp_barrier(CLK_LOCAL_MEM_FENCE | CLK_GLOBAL_MEM_FENCE);
+#elif defined(KOKKOS_ENABLE_HIP_ATOMICS)
+  __threadfence();
```
Comment (dalg24) [inline core/src/impl/Kokkos_Memory_Fence.hpp:58]: You didn't want the `memory_fence()` changes in a separate PR?

**Review by dalg24 — APPROVED** [review]

File: core/src/impl/Kokkos_Memory_Fence.hpp (hunk 1 above)
Comment (Rombur) [reply core/src/impl/Kokkos_Memory_Fence.hpp:58]: I greped where there was a difference between rocm macro and hip macro and that was one of them. I can split the commit.

Comment (dalg24) [reply core/src/impl/Kokkos_Memory_Fence.hpp:58]: Your choice.  Merge you are ready.

**Pull request merged**